frontend/src/App.tsx

```diff
@@ -38,21 +38,28 @@ const publicRoutes = [
 
 const loggedInRoutes = [
   { path: "/", Component: HomePage },
-  { path: "/create", Component: Add },
   { path: "/bank", Component: BankPage },
   { path: "/view/:id", Component: ViewQuestion, loader: qnLoader },
+  {path: "*", element: (<Navigate to="/" />)}
+]
+
+const adminOnlyRoutes = [
+  { path: "/create", Component: Add },
 ]
 
 
 function App() {
 
   const isAuthenticated = useSelector(selectIsAuthenticated);
-  const user = useSelector(selectUser)
+  const isAdmin = useSelector(selectIsAdmin)
 
-  console.log(user)
   console.log(isAuthenticated)
 
   const router = createBrowserRouter([
+    {
+      Component: NavbarWrapper,
+      children: isAuthenticated && isAdmin ? adminOnlyRoutes : []
+    },
     {
       Component: NavbarWrapper,
       children: isAuthenticated ? loggedInRoutes : publicRoutes
```

Comment:
I'm not sure if it is ideal to route like this. I feel like it might be better if we just list all the routes and adds a protect route to redirect the users.

```typescript
import { Navigate } from "react-router-dom";
// ...

export const AuthProtection = ({ children }) => {
  const user = // get user
  if (!user) {
    // user is not authenticated
    return <Navigate to="/" />;
  }
  return children;
};
```

```typescript
    element: (
      <AuthProtection>
        <ResgistrationPage />
      </AuthProtection>
    ),

// or
return <AuthProtection>
// ....
</AuthProtection>
```

Reply:
This might honestly be easier to understand. Will look into implementing protected routing like this

Reply:
I think this PR has gotten quite large, since this routing works for now, lets keep it this way.